README.mkd

```diff
@@ -242,6 +242,63 @@ As of November 2021 the additional tail-call entry points are only
 implemented in compiler-rt, and calls will only be generated by LLVM
 when the option `-mllvm -save-restore-tailcall` is specified.
 
+## Conventions for vendor extension
+
+Support for custom instruction set extensions are an important part of RISC-V,
+with large encoding spaces reserved of vendor extensions.
+
+However, there are no official guidelines on naming the mnemonics. This section
+defines guidelines which vendors are expected to follow if upstreaming support
+for their extensions. Although vendor-provided toolchains are free to make
+different choices, they are strongly urged to align with these guidelines in
+order to ensure there is a straightforward path for upstreaming in the future.
+
+### Vendor extension naming scheme
+
+According to the RISC-V ISA spec, non-standard extensions are named using a single `x`
+followed by an alphabetical name and an optional version number.
+
+To make it easier to identify and prevent naming conflict, vendor extensions
+should start with a vendor name, which could be an abbreviation of the full name.
+
+For example:
+- `XVentanaCondOps` from Ventana
+- `Xsfcflushdlone` from SiFive
+- `Xtheade` from T-head
+
+### Assembly mnemonic
+
+In order to avoid confusion between standard extension and other vendor
+extensions, instruction mnemonics from vendor extensions must have a
```

Comment:
Here the "must" should IMO also be replaced with a "should"

Reply:
I would stick with MUST here, as this is about namespace segmentation.

Open-source projects are always free to not adopt (or selectively adopt) this convention: so they aren't bound by the MUST unless they opt in.

However, we can incorporate this section by reference into a RISC-V policy (which would then be binding—for a suitably relaxed definition of "binding"—for our members) in the future if we make this a mandatory requirement here.

Reply:
I didn't have strong preference on that, so I leave the`must` here.

Reply:
FWIW I slightly prefer "should" here too. Marking anything in a "conventions" doc as a "must" seems like a logical contradiction to me. But obviously we're all coming to this with slightly different perceptions of what this doc is about!

It also feels a bit inconsistent to be heavy-handed with this rule when there are all sorts of other things vendors can do that are commonly not a good idea, but still allowable. e.g. using encoding space used by standard RISC-V extensions for their own purposes - normally a bad idea, but it's a freedom vendors have because there are times it might make sense.

But this isn't a blocking comment, it's not the biggest deal to me either way.